### PR TITLE
Feat/token list migration

### PR DESCRIPTION
**Pull request by fengtality**
**Before submitting this PR, please make sure**:

- [X] Your code builds clean without any errors or warnings
- [X] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:

Revised version of https://github.com/hummingbot/gateway/pull/155

**Tests performed by the developer**:

**Tips for QA testing**:

### CODE REVIEW

**Review by rapcmia — APPROVED** [review]
LGTM

- Revision of PR155, test latest commit 9451a6d2b7469b
- Run `yarn build` then check `conf/lists` folder, all ok ✅
   image
- Deleted `conf` folder and observe changes on `gateway-setup.sh`, all ok ✅ 
  image
- Ran tests on both local mbp(m1) and Ubuntu22.04 test server

**Comment by rapcmia** [comment]
PR update for docker build 
-  Create docker image for this PR and latest client development branch 

Using `hummigbot_gateway-compose`
- Use latest commit for deploy_examples repo 
- Setup hummingbot_gateway_compose file then `docker compose up -d`
    image
    - On the screenshot above, either `list` or `lists` is not available
    - Added docker compose file 👀 
      ```
      version: "3.9"
        services:
          hummingbot:
            container_name: hummingbot1
            image: hummingbot/hummingbot:dev
            volumes:
              - "./hummingbot_files/conf:/home/hummingbot/conf"
              - "./hummingbot_files/conf/connectors:/home/hummingbot/conf/connectors"
              - "./hummingbot_files/conf/strategies:/home/hummingbot/conf/strategies"
              - "./hummingbot_files/logs:/home/hummingbot/logs"
              - "./hummingbot_files/data:/home/hummingbot/data"
              - "./hummingbot_files/scripts:/home/hummingbot/scripts"
              - "./hummingbot_files/certs:/home/hummingbot/certs"
            # environment:
            #   - CONFIG_PASSWORD=test
            logging:
              driver: "json-file"
              options:
                  max-size: "10m"
                  max-file: "5"
            tty: true
            stdin_open: true
            network_mode: host
        
          gateway:
            container_name: gateway1
            image: hummingbot/gateway:159
            ports:
              - "15888:15888"
              - "8080:8080"
            volumes:
              - "./gateway_files/conf:/home/gateway/conf"
            #  - "./gateway_files/conf/lists:/home/gateway/conf/lists" 
              - "./gateway_files/logs:/home/gateway/logs"
              - "./gateway_files/db:/home/gateway/db"
              - "./hummingbot_files/certs:/home/gateway/certs"
            # environment:
            #   - GATEWAY_PASSPHRASE=a
      ```
      - Added `conf/lists` on the file for test purpose
      - Change folder permission as well and check `conf/lists` still empty

---

Ran tests using `gateway-create.sh`
- Follow the prompts and use PR docker image ✅ 
  image

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "license": "Apache-2.0",
   "repository": "https://github.com/hummingbot/gateway",
   "scripts": {
-    "prebuild": "rimraf dist && mkdir dist",
+    "prebuild": "rimraf dist && mkdir dist && yarn copy-files:token-list",
```
Comment (vic-en) [inline package.json:9]:
```suggestion
    "prebuild": "rimraf dist && mkdir dist",
```

File: package.json [hunk 2]
```diff
@@ -16,6 +16,7 @@
     "dev:debug": "NODE_ENV=dev DEBUG=* nodemon src/index.ts",
     "start": "/bin/bash ./startup.sh",
     "copy-files": "copyfiles -a 'src/**/schema/*.json' 'src/templates/*.yml' 'src/templates/lists/*.json' 'test/services/data/**/*.*' dist",
+    "copy-files:token-list": "mkdir -p conf/lists && copyfiles -f 'src/templates/lists/*.json' conf/lists",
```
Comment (vic-en) [inline package.json:19]:
```suggestion
    "copy-files:token-list": "mkdir -p conf/lists && copyfiles -f -s 'src/templates/lists/*.json' conf/lists",
```

File: package.json [hunk 3]
```diff
@@ -16,6 +16,7 @@
     "dev:debug": "NODE_ENV=dev DEBUG=* nodemon src/index.ts",
     "start": "/bin/bash ./startup.sh",
```
Comment (vic-en) [inline package.json:17]:
```suggestion
    "start": "yarn copy-files:token-list && /bin/bash ./startup.sh",
```

**Review by vic-en — CHANGES_REQUESTED** [review]
@fengtality 
Docker and source will work after those suggestions are commited.

1. It's advisable to remove the `copy_lists` function in the gateway-setup.sh files.
2. [Set 3](https://github.com/hummingbot/deploy-examples/blob/main/hummingbot_gateway_compose/README.md#3-add-token-lists)  of docker compose file for gateway can be removed as well.

I couldn't push commits because the pr was re-created.

**Comment by fengtality** [comment]
Closed in favor of #160

**Pull request closed without merging**